### ISSUE TITLE
UndefVarError: tmp_lab not defined

### ISSUE THREAD

**Issue by Tyde** [JuliaDiffEq/DiffEqBase.jl#27]
When I try to plot a solution of an `ODEProblem` I do get the following error 

```
UndefVarError: tmp_lab not defined

 in add_labels!(::Array{String,1}, ::Tuple{DiffEqBase.##118#119,Int64,Int64}, ::Int64, ::DiffEqBase.ODESolution{Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},DiffEqBase.ODEProblem{Array{Float64,2},Float64,false,#f},OrdinaryDiffEq.Tsit5,OrdinaryDiffEq.InterpolationData{#f,Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}}) at C:\Users\thiem\.julia\v0.5\DiffEqBase\src\solutions\solution_interface.jl:231
 in solplot_vecs_and_labels(::Int64, ::Array{Tuple,1}, ::Array{Array{Float64,2},1}, ::Array{Float64,1}, ::DiffEqBase.ODESolution{Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},DiffEqBase.ODEProblem{Array{Float64,2},Float64,false,#f},OrdinaryDiffEq.Tsit5,OrdinaryDiffEq.InterpolationData{#f,Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}}, ::Bool) at C:\Users\thiem\.julia\v0.5\DiffEqBase\src\solutions\solution_interface.jl:295
 in macro expansion at C:\Users\thiem\.julia\v0.5\DiffEqBase\src\solutions\solution_interface.jl:84 [inlined]
 in apply_recipe(::Dict{Symbol,Any}, ::DiffEqBase.ODESolution{Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},DiffEqBase.ODEProblem{Array{Float64,2},Float64,false,#f},OrdinaryDiffEq.Tsit5,OrdinaryDiffEq.InterpolationData{#f,Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}}) at C:\Users\thiem\.julia\v0.5\RecipesBase\src\RecipesBase.jl:238
 in _process_userrecipes(::Plots.Plot{Plots.GRBackend}, ::Dict{Symbol,Any}, ::Tuple{DiffEqBase.ODESolution{Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},DiffEqBase.ODEProblem{Array{Float64,2},Float64,false,#f},OrdinaryDiffEq.Tsit5,OrdinaryDiffEq.InterpolationData{#f,Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}}}) at C:\Users\thiem\.julia\v0.5\Plots\src\pipeline.jl:73
 in _plot!(::Plots.Plot{Plots.GRBackend}, ::Dict{Symbol,Any}, ::Tuple{DiffEqBase.ODESolution{Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},DiffEqBase.ODEProblem{Array{Float64,2},Float64,false,#f},OrdinaryDiffEq.Tsit5,OrdinaryDiffEq.InterpolationData{#f,Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}}}) at C:\Users\thiem\.julia\v0.5\Plots\src\plot.jl:171
 in plot(::DiffEqBase.ODESolution{Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},DiffEqBase.ODEProblem{Array{Float64,2},Float64,false,#f},OrdinaryDiffEq.Tsit5,OrdinaryDiffEq.InterpolationData{#f,Array{Array{Float64,2},1},Array{Float64,1},Array{Array{Array{Float64,2},1},1},OrdinaryDiffEq.Tsit5ConstantCache{Float64,Float64}}}) at C:\Users\thiem\.julia\v0.5\Plots\src\plot.jl:46
```

The example code I am using is the following:

```
using DifferentialEquations,Plots
A = [1. 0 0 -5
     4 -2 4 -3
     -4 0 0 1
     5 -2 2 3]
u0 = rand(4,2)
tspan = (0.0,1.0)
f(t,u) = A * u
prob = ODEProblem(f,u0,tspan)
sol = solve(prob)
plot(sol)
```

**Comment by ChrisRackauckas** [comment]
Yes, I realized that there was a problem in the DiffEqBase.jl v0.14.0 release. It's from when I [added the new function plotting stuff](http://docs.juliadiffeq.org/latest/basics/plot.html#Choosing-Variables-1) (see you can choose a function now... but that messed up the labels). The labels are break when plotting multiple variables at the same time. 

```julia
plot(sol,vars=(0,1)) # Works
plot(sol,vars=[(0,1),(0,2)]) # Fails
```

Sorry about that! It's fixed on master, but master has quite a few other changes so it's best to avoid that until it's released (it's the big "pre-workshop, do all the API changes now" update). I would suggest pinning to v0.13.0 for now:

```julia
Pkg.pin("DiffEqBase",v"0.13.0")
```

I am sorry for the error! This was really bad and disruptive... I am really sorry I didn't catch it until last night (plotting doesn't have unit tests so I have to check it by hand... and I only checked with single timeseries). I hope to have the major release METADATA PRs out today.

On this note, I should think about adding VisualRegressionTests.jl now that the plotting is "finalized"...

**Comment by ChrisRackauckas** [comment]
The first tags are in, but this might take a day or two since:

StochasticDiffEq
OrdinaryDiffEq
DiffEqDevTools
DiffEqProblemLibrary
DiffEqPDEBase
DelayDiffEq
FiniteElementDiffEq
Sundials
ODEInterfaceDiffEq
ODE
LSODA

all of those packages need to update before Pkg will pull DiffEqBase 0.15.0 (because of maximum version bounds due to incompatibility). 

That said... lots of good stuff is in there.

**Comment by ChrisRackauckas** [comment]
fixed by 0.15.0. You will need to wait for the other tags to go through in order for this to not be auto-downgraded by Pkg.update() though.

**Comment by ChrisRackauckas** [comment]
You should be able to `Pkg.update()` and have it work now. Thanks for the bug report!
